array_api_compat/torch/_aliases.py

```diff
@@ -602,6 +606,43 @@ def tensordot(x1: array, x2: array, /, *, axes: Union[int, Tuple[Sequence[int],
     x1, x2 = _fix_promotion(x1, x2, only_scalar=False)
     return torch.tensordot(x1, x2, dims=axes, **kwargs)
 
+
+def isdtype(
+    dtype: Dtype, kind: Union[Dtype, str, Tuple[Union[Dtype, str], ...]],
+    *, _tuple=True, # Disallow nested tuples
+) -> bool:
+    """
+    Returns a boolean indicating whether a provided dtype is of a specified data type ``kind``.
+
+    Note that outside of this function, this compat library does not yet fully
+    support complex numbers.
+
+    See
+    https://data-apis.org/array-api/latest/API_specification/generated/array_api.isdtype.html
+    for more details
+    """
+    if isinstance(kind, tuple) and _tuple:
+        return builtin_any(isdtype(dtype, k, _tuple=False) for k in kind)
+    elif isinstance(kind, str):
+        if kind == 'bool':
+            return dtype == torch.bool
+        elif kind == 'signed integer':
+            return dtype in _int_dtypes and dtype.is_signed
```

Comment:
Is there a better way to test if a torch dtype is integral.

Reply:
There isn't. Based on dtype properties it would be:
```python
not dtype.is_floating_point and not dtype.is_complex and dtype.is_signed and dtype != dtype.bool
```
which looks worse and may also not be foolproof. The check you have will need explicit updating if for example `uint16` is added to PyTorch, but that's okay (nothing like that is in the pipeline AFAIK).

Reply:
Hopefully this function itself would be added to pytorch by then.